app/src/main/java/com/android/sample/model/profile/ProfileViewModel.kt

```diff
@@ -115,4 +122,21 @@ constructor(private val repository: ProfilesRepository, private val localDatabas
   fun loadCachedProfile(): User? {
     return runBlocking { localDatabase.userDao().getUser(Firebase.auth.currentUser?.uid ?: "") }
   }
+  /** Check if the activity should be displayed based on the category and the user's role in the */
+  fun shouldShowActivity(activity: Activity, user: User, category: String): Boolean {
+    return when (category) {
+      "created" -> activity.creator == user.id && activity.date > Timestamp.now()
+      "enrolled" -> activity.creator != user.id && activity.date > Timestamp.now()
+      "past" -> activity.date < Timestamp.now()
+      else -> false
+    }
+  }
```

Comment:
We just have to update this to check for the exact time instead of the date, but as it is my task, I will not require any changes